modules/xfeatures2d/src/latch.cpp

```diff
@@ -500,20 +500,21 @@ namespace cv
             if ( keypoints.empty() )
                 return;
 
-
             Mat grayImage;
-            GaussianBlur(image, grayImage, cv::Size(3, 3), 2, 2);
-
-            if (image.type() != CV_8U) cvtColor(image, grayImage, COLOR_BGR2GRAY);
```

Comment:
I would simply reposition line 505 to line 512 and further! In that case the blur is added in both color and grayscale image case.